src/main/kotlin/pl/put/smartgarden/domain/device/SensorService.kt

```diff
@@ -1,10 +1,63 @@
 package pl.put.smartgarden.domain.device
 
 import org.springframework.stereotype.Service
+import pl.put.smartgarden.domain.device.SensorType.HUMIDITY
+import pl.put.smartgarden.domain.device.SensorType.ILLUMINANCE
+import pl.put.smartgarden.domain.device.SensorType.IRRIGATION
+import pl.put.smartgarden.domain.device.SensorType.TEMPERATURE
+import pl.put.smartgarden.domain.device.dto.request.SensorRequest
+import pl.put.smartgarden.domain.device.exception.SensorInAnotherDeviceException
+import pl.put.smartgarden.domain.device.repository.SensorRepository
 
 @Service
-class SensorService {
-    fun createSensors() {
-        TODO("Not yet implemented")
+class SensorService(val sensorRepository: SensorRepository) {
+    fun createSensors(deviceId: Int, sensorsRequest: List<SensorRequest>): List<Sensor> {
+        val sensorsInDb = sensorRepository.findAllByDeviceId(deviceId)
+        val requestGuids = sensorsRequest.map { it.guid }
+        checkIfSensorsAreUsedInAnotherDevice(requestGuids, deviceId)
+
+        val inactiveSensors = inactivateNotUsedSensors(sensorsInDb, requestGuids)
```

Comment:
Nie jestem pewien czy dobrze rozumiem, ale inactivateNotUsedSensors() zwraca chyba tylko aktualnie dezaktywowane sensory, te które już wcześniej były nieaktywne nie są zwracane. Czy to zamierzone?

PS. można dodać komentarze nad tymi metodami z informacją co zwracają - bo każda zwraca typ List<Sensor> ale trzeba spojrzeć w kod żeby się dowiedzieć co konkretnie :)

Reply:
Dokładnie tak to działa - pokazuje, co zostało zmienione w bazie tak jakby